cloud/smartdatacenter/sdc.py

```diff
@@ -0,0 +1,316 @@
+    description:
+      - Comma separated list of desired networks ids, obtained from ListNetworks
+    required: false
+  count:
+    description:
+      - An integer value which indicates how many instances that should be created. Only used if count_tag or exact_count are missing (not recommended).
+    required: false
+    default: 1
+    aliases: []
+  exact_count:
+    description:
+      - An integer value which indicates how many instances that match the 'count_tag' parameter should be running.
+    required: false
+    default: null
+    aliases: []
+  count_tag:
+    description:
+      - Used with 'exact_count' to determine how many nodes based on a specific tag criteria should be running.  This can be expressed in multiple ways and is shown in the EXAMPLES section.  For instance, one can request 25 servers that are tagged with "class=webserver". Count tag(s) will always be added to created machines
+    required: false
+    default: null
+    aliases: []
+  tags:
+    description:
+      - Tags to assign to this instance.
+    required: false
+    default: null
+    aliases: []
+  wait:
+    description:
+      - wait for the instance to be in state 'running' before returning
+    required: false
+    default: "yes"
+    choices: [ "yes", "no" ]
+    aliases: []
+  wait_timeout:
+    description:
+      - how long before wait gives up, in seconds
+    default: 600
+    aliases: []
+  state:
+    description:
+      - create or terminate instances
+    required: false
+    default: 'present'
+    aliases: []
+
+requirements: [ "smartdc" ]
+author: Preston Marshall
+'''
+
+EXAMPLES = '''
+
+# Provision single machine example
+- local_action:
+    module: sdc
+    account: myaccount
+    key_id: /myaccount/keys/mykey
+    name: my-virtual-machine
+    package: 486bb054-6a97-4ba3-97b7-2413d5f8e849
+    image: 62f148f8-6e84-11e4-82c5-efca60348b9f
+    location: 'us-east-1.api.joyentcloud.com'
+
+# Ensure 5 machines with tag "webserver" are running, does not destroy existing machines
+- local_action:
+    module: sdc
+    account: myaccount
+    key_id: /myaccount/keys/mykey
+    package: 486bb054-6a97-4ba3-97b7-2413d5f8e849
+    image: 62f148f8-6e84-11e4-82c5-efca60348b9f
+    location: 'us-east-1.api.joyentcloud.com'
+    count_tag:
+      role: webserver
+    exact_count: 5
+
+# Terminate machines by their tags
+- local_action:
+    module: sdc
+    account: myaccount
+    key_id: /myaccount/keys/mykey
+    location: 'us-east-1.api.joyentcloud.com'
+    tags:
+      role: webserver
+    state: absent
+
+# Terminate machine by its id
+- local_action:
+    module: sdc
+    account: myaccount
+    key_id: /myaccount/keys/mykey
+    machine_id: 51307f2f-02a3-4bdc-9b38-2f263eb43e2d
+    location: 'us-east-1.api.joyentcloud.com'
+    state: absent
+'''
+
+import os
+import sys
+import time
+try:
+    import smartdc
+    from smartdc import DataCenter
+except ImportError:
+    print "failed=True msg='smartdc required for this module'"
```

Comment:
This form of 'Baby JSON' was removed from ansible 1.8.  The recommended pattern is something like:

```python
try:
    import smartdc
    HAS_SMARTDC = True
except ImportError:
    HAS_SMARTDC = False

# Other code

def main():
    module = AnsibleModule(...)
    if not HAS_SMARTDC:
        module.fail_json(msg='smartdc python module is required')
```